admin-dev/themes/new-theme/js/components/pagination/dynamic-paginator.js

```diff
@@ -0,0 +1,298 @@
+        this.paginate(page);
+      }
+    });
+    this.$paginationContainer.on('change', this.selectorsMap.limitSelect, () => {
+      this.paginate(1);
+    });
+  }
+
+  /**
+   * @param {Number} page
+   */
+  async paginate(page) {
+    const limit = this.getLimit();
+    const data = await this.dataProvider.fetch(this.calculateOffset(page, limit), limit);
+    $(this.selectorsMap.jumpToPageInput).val(page);
+    this.countPages(data.total);
+    this.refreshButtonsData(page);
+    this.refreshInfoLabel(page, data.total);
+
+    this.toggleFirstPageAvailability(page > 1);
+    this.togglePreviousPageAvailability(page > 1);
+    this.toggleNextPageAvailability(page < this.pagesCount);
+    this.toggleLastPageAvailability(page < this.pagesCount);
+
+    this.renderer.render(data);
+  }
+
+  /**
+   * @param page
+   * @param limit
+   *
+   * @returns {Number}
+   */
+  calculateOffset(page, limit) {
+    return (page === 1) ? 0 : (page - 1) * limit;
+  }
+
+  /**
+   * @param {Number} page
+   *
+   * @private
+   */
+  refreshButtonsData(page) {
+    this.$paginationContainer.find(this.selectorsMap.nextPageBtn).data('page', page + 1);
+    this.$paginationContainer.find(this.selectorsMap.previousPageBtn).data('page', page - 1);
+    this.$paginationContainer.find(this.selectorsMap.lastPageBtn).data('page', this.pagesCount);
+  }
+
+  /**
+   * @param {Number} page
+   * @param {Number} total
+   */
+  refreshInfoLabel(page, total) {
+    const infoLabel = this.$paginationContainer.find(this.selectorsMap.paginationInfoLabel);
+    const limit = this.getLimit();
+    const from = page === 1 ? 1 : Math.round((page - 1) * limit);
+    const to = page === this.pagesCount ? total : Math.round(page * limit);
+    const modifiedInfoText = infoLabel.data('pagination-info')
+      .replace(/%from%/g, from)
+      .replace(/%to%/g, to)
+      .replace(/%total%/g, total)
+      .replace(/%current_page%/g, page)
+      .replace(/%page_count%/g, this.pagesCount);
+
+    infoLabel.text(modifiedInfoText);
+  }
+
+  /**
+   * @param {Boolean} enable
+   *
+   * @private
+   */
+  toggleFirstPageAvailability(enable) {
+    this.toggleTargetAvailability(this.$paginationContainer.find(this.selectorsMap.firstPageItem), enable);
+  }
+
+  /**
+   * @param {Boolean} enable
+   *
+   * @private
+   */
+  toggleLastPageAvailability(enable) {
+    this.toggleTargetAvailability(this.$paginationContainer.find(this.selectorsMap.lastPageItem), enable);
+  }
+
+  /**
+   * @param {Boolean} enable
+   *
+   * @private
+   */
+  togglePreviousPageAvailability(enable) {
+    this.toggleTargetAvailability(this.$paginationContainer.find(this.selectorsMap.previousPageItem), enable);
+  }
+
+  /**
+   * @param {Boolean} enable
+   *
+   * @private
+   */
+  toggleNextPageAvailability(enable) {
+    this.toggleTargetAvailability(this.$paginationContainer.find(this.selectorsMap.nextPageItem), enable);
+  }
```

Comment:
Definitly I don't get it, having this.selectorsMap.previousPageItem is verbose enough to me

Reply:
I'll simplify this by moving the `find` part in the `toggleTargetAvailability` function